src/main/java/com/samoilenko/onlinebookstore/service/impl/BookServiceImpl.java

```diff
@@ -37,4 +37,16 @@ public BookDto findBookById(Long id) {
         ));
     }
 
+    @Override
+    public void deleteById(Long id) {
+        bookRepository.deleteById(id);
+    }
+
+    @Override
+    public BookDto update(Long id, BookRequestDto bookRequestDto) {
+        Book updatedBook = bookMapper.toModel(bookRequestDto);
+        updatedBook.setId(id);
+        return bookMapper.toDto(bookRepository.save(updatedBook));
+    }
```

Comment:
first, we need to check if the book with such an id exists, if not - throw `EntityNotFountException`, and than act, fix for both methods